pydantic/_internal/_generics.py

```diff
@@ -113,6 +114,14 @@ def __delitem__(self, key: KT) -> None:
 _GENERIC_TYPES_CACHE = GenericTypesCache()
 
 
+class PydanticGenericMetadata(typing_extensions.TypedDict, total=False):
+    args: tuple[Any, ...] | None
+    defaults: dict[str, Any] | None
```

Comment:
Some docstrings would be nice for these
```suggestion
    args: tuple[Any, ...] | None
    """Generic arguments to this model, e.g. `Model[int, str]` -> `(int, str)`"""
    defaults: dict[str, Any] | None
```

Reply:
since it's just args, origin, and parameters now, I just added comments saying `# analogous to typing._GenericAlias.__args__` etc. Let me know if you think that's insufficient.